junit-platform/build.gradle

```
@@ -1,5 +1,5 @@
 plugins {
-    id "java"
+    id "java-library" //historically publishes blockhound as compile dependency in pom
```

this sounds like something we're considering changing and is there only for legacy reasons, but (unless there are good reasons that should be discussed separately) this is not the case.
```suggestion
    id "java-library"
```